src/js/Helpers/Layover.js

```diff
@@ -0,0 +1,868 @@
+    } else {
+      toggleHeight = this._toggle.offsetHeight;
+      toggleWidth = this._toggle.offsetWidth;
+    }
+
+    let addToTop = 0;
+    let addToLeft = 0;
+    if (!vp.top || !vp.bottom) {
+      const multiplier = vp.top ? -1 : 1;
+      if (!vp.bottom && y === VerticalAnchors.OVERLAP) {
+        addToTop += toggleHeight;
+      } else if (y === VerticalAnchors.TOP || y === VerticalAnchors.BOTTOM) {
+        addToTop += (multiplier * toggleHeight);
+      }
+
+      addToTop += (multiplier * childHeight);
+
+      this._lastYFix = vp.top ? 'bottom' : 'top';
+    }
+
+    if (x !== HorizontalAnchors.CENTER && (!vp.left || !vp.right)) {
+      if (!vp.left && x === HorizontalAnchors.LEFT) {
+        addToLeft += toggleWidth + childWidth;
+        this._lastXFix = 'left';
+      } else if (!vp.left && x === HorizontalAnchors.INNER_LEFT) {
+        addToLeft += toggleWidth;
+        this._lastXFix = 'left';
+      } else if (!vp.right && x === HorizontalAnchors.RIGHT) {
+        addToLeft -= (toggleWidth + childWidth);
+        this._lastXFix = 'right';
+      } else if (!vp.right && x === HorizontalAnchors.INNER_RIGHT) {
+        addToLeft -= toggleWidth;
+        this._lastXFix = 'right';
+      }
+    }
+
+
+    if (addToTop !== 0 || addToLeft !== 0) {
+      this._initialTop += addToTop;
+      this._initialLeft += addToLeft;
+
+      this.setState({ styles: this._mergeStyles({ top: this._initialTop, left: this._initialLeft }) });
+    }
+  };
+
+  /**
+   * When the child is initially mounted, it will update the styles for centering
+   * the element (if enabled) and then attempt to fix any viewport issues.
+   */
+  _fixateChild = (child) => {
+    this._child = findDOMNode(child);
+
+    if (this._child !== null) {
+      const { children, centered } = this.props;
+      const anchor = this._getAnchor(this.props);
+      this._childComponent = React.Children.only(children);
+
+      // If child also has a ref callback, simulate the same thing
+      if (typeof this._childComponent.ref === 'function') {
+        this._childComponent.ref(child);
+      }
+
+      if (!this._child || (!this._toggle && !this._contextRect)) {
+        return;
+      }
+
+      if (this._dialog && this._dialog.classList.contains('md-dialog--centered')) {
+        return;
+      }
+
+      const rect = this._contextRect || this._toggle.getBoundingClientRect();
+      const styles = this._createStyles(anchor, centered, this._child, rect);
+      if (styles.top || styles.left) {
+        this._initialLeft = styles.left || this._initialLeft;
+        this._initialTop = styles.top || this._initialTop;
+        this.setState({ styles: this._mergeStyles(styles) }, this._initialFix);
+      } else {
+        this._initialFix();
+      }
+    } else if (this._childComponent && typeof this._childComponent.ref === 'function') {
+      this._childComponent.ref(child);
+    }
+  };
+
+  _handleScroll = () => {
+    if (!this._ticking) {
```

Comment:
Should implement a trailing version of this. Scrolling fast sometimes doesn't have last calculation accurate enough.